azkaban-web-server/src/main/java/azkaban/webapp/servlet/LoginAbstractAzkabanServlet.java

```diff
@@ -396,6 +395,30 @@ protected boolean hasPermission(final Project project, final User user,
     return false;
   }
 
+  protected Project filterProjectByPermission(final Project project, final User user,
```

Comment:
Please add some comments for the new functions.

Reply:
addressed in coming change.

Reply:
addressed